README.md

```diff
@@ -1,9 +1,126 @@
+    further analysis, e.g. trained models, model predictions on test data sets
+  - Doesn't enforce any constraints or requirements on your models' interfaces
+  - Pure Python package, with no other setup or configuration of other systems required
+
+## Usage
+
+Import your packages for modelling and kotsu:
+
+```python
+from sklearn import datasets, svm
+from sklearn.model_selection import cross_val_score
+
+import kotsu
+```
+
+Register your competing models. Here we register two Support Vector Classifiers with different
+hyper-parameters:
+
+```python
+model_registry = kotsu.registration.ModelRegistry()
+
+model_registry.register(
+    id="SCV-v1",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 1, "random_state": 1},
+)
+
+model_registry.register(
+    id="SCV-v2",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 0.5, "random_state": 1},
+)
+```
+
+Register your validations. You can register multiple if you want to compare models over different
+tests, e.g. on different datasets. Your validation should take an instance of your models as an
+argument and return a dictionary containing the results. Here we register two Cross-Validation
+validations with different numbers of folds:
+
+```python
+validation_registry = kotsu.registration.ValidationRegistry()
+
+
+def factory_iris_cross_valdation(folds: int):
+    """Factory for iris cross validation."""
+
+    def iris_cross_valdation(model) -> dict:
+        """Iris classification cross validation."""
+        X, y = datasets.load_iris(return_X_y=True)
+        scores = cross_val_score(model, X, y, cv=folds)
+        results = {f"fold_{i}_score": score for i, score in enumerate(scores)}
+        results["mean_score"] = scores.mean()
+        results["std_score"] = scores.std()
+        return results
+
+    return iris_cross_val
+
+
+validation_registry.register(
+    id="iris_cross_val-v1",
+    entry_point=factory_iris_cross_valdation,
+    kwargs={"folds": 5},
+)
+
+validation_registry.register(
+    id="iris_cross_val-v2",
+    entry_point=factory_iris_cross_valdation,
+    kwargs={"folds": 10},
+)
+```
+
+Run the models through the validations, choosing the current directory as location for the output
+of the results:
+
+```python
+kotsu.run(model_registry, validation_registry, "./")
+```
+
+Then find the results from each model-validation combination in a dataframe written to the current
+directory.
+
+#### Documentation on interfaces
+
+See [kotsu.typing](https://github.com/datavaluepeople/kotsu/blob/main/kotsu/typing.py) for
+documentation on the main entities; Models, Validations, and Results, and their interfaces.
+
+#### Comprehensive example
+
+See the [end to end test](https://github.com/datavaluepeople/kotsu/blob/main/tests/test_end_to_end.py)
+for a more comprehensive example usage of kotsu, which includes storing the trained models from
+each model-validation run.
+
+## Where to get it
```

Comment:
I think the `Installation` being right at the top of the README (a la [truman](https://github.com/datavaluepeople/truman)) is maybe a bit better? what do you think

Reply:
I would agree with Ali on moving this up to the top.

Reply:
3e29ad9